Source/WebCore/style/StyleResolver.cpp

```diff
@@ -287,6 +288,10 @@ auto Resolver::initializeStateAndStyle(const Element& element, const ResolutionC
         state.setParentStyle(RenderStyle::clonePtr(*state.style()));
     }
 
+    // BuilderState::useSVGZoomRulesForLength equivalent
+    if (is<SVGElement>(element) && !(is<SVGSVGElement>(element) && element.parentNode()))
+        state.style()->setUseSVGZoomRulesForLength(true);
+
```

Comment:
Lets do this sort of thing from the `Style::Builder` where it can be just `m_state.style().setUseSVGZoomRulesForLength(m_state.useSVGZoomRulesForLength())`.